src/main/java/networkmanagementsystem/NetworkManagementSystem.java

```diff
@@ -35,444 +35,418 @@
+
+    /**
+     * Handle the request to perform self diagnostics on a radio unit.
+     *
+     * @param ip The IP address of the radio unit that will perform self diagnostics.
+     */
+    @Override
+    public void performSelfDiagnostics(String ip) {
+        List<Object> params = new ArrayList<>(Arrays.asList(ip, ProcedureOptions.DIAGNOSTIC));
+        support.firePropertyChange(Procedure.COMMISSION.getDesc(), ProcedureOptions.RAT_TYPE, params);
+    }
+
+    /**
+     * Handle the request to remove all carriers from a radio unit.
+     *
+     * @param ip The IP address of the radio unit that will have all of its carriers removed.
+     */
+    @Override
+    public void removeAllCarriersOnRu(String ip) {
+        List<Object> params = new ArrayList<>(Arrays.asList(ip, ProcedureOptions.CARRIER));
+        support.firePropertyChange(Procedure.DECOMMISSION.getDesc(), ProcedureOptions.RAT_TYPE, params);
+    }
+
+    /**
+     * Handle the request to list all created radio units.
+     */
+    @Override
+    public void listNetworkInventory() {
+        support.firePropertyChange(Procedure.LIST.getDesc(), ProcedureOptions.FULL, "");
+    }
+
+    /**
+     * Handle the request to list all radio units that contain a certain parameter.
+     *
+     * @param param The parameter that will be used to query for radio units. Examples
+     *              include a RatType, a RadioUnitState, a FrequencyBand, or a String name.
+     */
+    @Override
+    public void listRuByParam(Object param) {
+        support.firePropertyChange(Procedure.LIST.getDesc(), ProcedureOptions.PARAM, param);
+    }
+
+    /**
+     * Handle the request to list the details of a radio unit.
+     *
+     * @param ip The IP address of the radio unit that will have its details listed.
+     */
+    @Override
+    public void listRadioUnitDetails(String ip) {
+        support.firePropertyChange(Procedure.LIST.getDesc(), ProcedureOptions.RU, ip);
+    }
+
+    /**
+     * Handle the request to set an alarm on a radio unit.
+     *
+     * @param ip    The IP address of the radio unit that will have an alarm raised.
+     * @param alarm The alarm that will be raised on the radio unit.
+     */
+    @Override
+    public void setAlarmOnRu(String ip, AlarmStatusLevel alarm) {
+        List<Object> params = new ArrayList<>(Arrays.asList(ip, alarm));
+        support.firePropertyChange(Procedure.CREATE.getDesc(), ProcedureOptions.ALARM, params);
+    }
+
+    /**
+     * Handle the request to list all current network alarms.
+     */
+    @Override
+    public void getNetworkAlarms() {
+        List<ManagedRadioUnit> radioUnits = radioUnitRegistry.getAllRadios();
+
+        radioUnits = radioUnits.stream()
+                .filter(ru -> ru.getAlarmStatus() != AlarmStatusLevel.NO_ALARM)
+                .collect(Collectors.toList());
+
+        if (radioUnits.size() > 0) {
+        	radioUnits.forEach(ru -> System.out.println(ru.getIpAddress() + ": " + ru.getAlarmStatus() + "\n"));

```

Comment:
I was experiencing some strangeness when printing from this thread, when this was called from the alarm monitoring thread - specifically that the console would immediately swap to this thread. May not be that big of an issue, but the original implementation of returning a string to the caller (either the other thread or the client) was to get around that. Perhaps that's just something from my IDE though?

Reply:
I was wondering about this as well, but now that you've explained it that makes sense. I think we could revert this change back then before submission.